added polling for window.BrowserStack and correct order of tape output

youzi opened the pull request:
Tape output wasn't captured when window.BrowserStack is not initialised, causing missing output and often hanging tests (never post results). Solved this by polling (500ms).

Order of tape output was not guaranteed. Solved this by queueing post requests.

Commit:
added polling for window.BrowserStack and correct order of tape output

Commit:
dont timeout if BrowserStack is already there

Commit:
added missing cb

Commit:
guard for falsy string

Commit:
added example and concat log body for nicer TAP output

Commit:
calling .getHarnass later somehow allows for better async logs

Commit:
added devDependency for tap-difflet (for pretty tap output)

Commit:
prettier output

TehShrike commented:
Oh man I'm so happy to see this pull request.  The missing output/hanging has been seriously annoying me.

I'm at a conference right now so it might be a few days until I can spend more than 10 minutes looking at this.

I'm going to have to remind myself the impact that the `tape.createStream()` call has on everything else (does it not change the behavior of the tape harness?), and I'll try running the tests locally, but I'll merge/publish this as soon as I can.

Thank you!

TehShrike commented on package.json:
```diff
@@ -4,8 +4,7 @@
   "scripts": {
-    "start": "browserify index.js example/index.js -o example/bundle.js && browserstack-runner",
-    "local": "browserify index.js example/index.js -o example/bundle.js && node example/bundle.js",
+    "start": "browserify index.js example/index.js -o example/bundle.js && browserstack-runner | tap-difflet",
```
`start` implies a real application-type script for me - I'd be fine with replacing the `test` script with this same thing, since it really is a development test.  I'd be down with renaming the `example` folder to `test`, too.  (Thanks for writing an actual test by the way!)

TehShrike commented on browserstack.json:
```diff
@@ -0,0 +1,19 @@
+{
```
Is there any way to get this file into the `example`/`test` folder?  Having it in the root directory implies that it matters to normal usage of the library.  If that turns out to be difficult, it's not a big deal.

TehShrike commented on index.js:
```diff
@@ -25,6 +24,7 @@ function postLogs () {
 }
 
 function postReport () {
+  var results = tape.getHarness()._results
```
> calling .getHarnass later somehow allows for better async logs

lol that's scary

TehShrike commented:
You pushed up a couple more codes since I last looked - how are you feeling about it now?  Want me to test/publish soon, or do you want to poke at it some more?  Let me know.

And again, thanks so much for this patch!  This will make my life noticeably better.

youzi commented:
I updated a couple of more times:
- Concatting all tape output into a single log, so we can have proper TAP output (browserstack-runner adds prefixes to every log :( ). This way we also don't have to queue the logs one by one, making it faster.
- Added an example (not sure if this is a test, since it doesn't actually test the module, it shows an example of a test using the module)
- Fixed some bugs

I'm pretty confident it's ready to publish. Me and others will be working with this intensively over the next days, so I can give you a shout when I think you can publish.

I'll go ahead and update the dir structure.

Commit:
moved browserstack.json to example dir

Commit:
added failing test example

TehShrike commented:
Feeling good?

TehShrike commented:
Works for me.  Thanks!

TehShrike commented:
Published as 1.1.0

youzi commented:
@TehShrike Thanks for merging. Apologies for the radio silence, we've been using it and it's been working fine for us.

TehShrike commented:
No problem, thanks so much for the help!
